pytorch_lightning/lite/lite.py

```diff
@@ -460,6 +476,27 @@ def _check_strategy_support(self, strategy: Optional[Union[str, Strategy]]) -> N
                 f" Choose one of {supported} or pass in a `Strategy` instance."
             )
 
+    def _check_accelerator_type(self, accelerator: Accelerator) -> None:
+        if not isinstance(accelerator, self._supported_accelerators()):
+            supported_values = ["auto"] + [x.lower() for x in self._supported_device_types]
+            raise MisconfigurationException(
+                f"`accelerator={repr(accelerator)}` is not a valid choice for LightningLite."
+                f" Choose one of {supported_values} or pass in a `Accelerator` instance."
+            )
+
+    def _check_strategy_type(self, strategy: Optional[Union[str, Strategy]]) -> None:
+
+        if not isinstance(strategy, self._supported_strategies()):
+            valid = [t.lower() for t in self._supported_strategy_types()]
+            raise MisconfigurationException(
+                f"`strategy={repr(strategy)}` is not a valid choice."
+                f" Choose one of {valid} or pass in a `Strategy` instance."
+            )
+
+    @staticmethod
+    def _supported_accelerators() -> Tuple[Type[Accelerator], ...]:
+        return (CPUAccelerator, GPUAccelerator, TPUAccelerator)
+
     @staticmethod
     def _supported_device_types() -> Sequence[_AcceleratorType]:
```

Comment:
Can't we remove this and `_supported_strategy_types` now?

Reply:
short answer: no, See https://github.com/PyTorchLightning/pytorch-lightning/pull/12069#discussion_r815325126

long answer:
no, they are still necessary for input validation unfortunately. If this is removed, the input validation can only be done after the accelerator and strategy init and in case of unsupported arguments, the accelerator connector may error out with rather hard to interpret key errors or fall back to cpu in which case the user won't get any notification on this.